cpp/src/table/row_operators.cu

```diff
@@ -255,6 +255,24 @@ auto decompose_structs(table_view table,
                          std::move(verticalized_col_depths));
 }
 
+/*
+ * This helper function generates dremel data for any list-type columns in a
+ * table. This data is necessary for lexicographic comparisons.
+ */
+auto list_lex_preprocess(table_view table, rmm::cuda_stream_view stream)
+{
+  std::vector<detail::dremel_data> dremel_data;
+  std::vector<detail::dremel_device_view> dremel_device_views;
+  for (auto const& col : table) {
+    if (col.type().id() == type_id::LIST) {
+      dremel_data.push_back(detail::get_dremel_data(col, {}, false, stream));
+      dremel_device_views.push_back(dremel_data.back());
+    }
+  }
+  auto d_dremel_device_views = detail::make_device_uvector_async(dremel_device_views, stream);
```

Comment:
`dremel_device_views` only has a lifetime as long as this function scope. If we make an `rmm::device_uvector` asynchronously, how do we know that the host data is still alive when the device is ready to copy it? A stream sync or synchronous copy may be required.

Reply:
Great catch, I agree that this needs to be a synchronous construction.